yuniql-core/MigrationServiceBase.cs

```diff
@@ -1,244 +1,316 @@
+                $"{string.Join(Environment.NewLine, sqlScriptFiles.Select(s => "  + " + new FileInfo(s).Name))}");
 
             //execute all script files in the target folder
-            sqlScriptFiles.Sort();
-            sqlScriptFiles.ForEach(scriptFile =>
-            {
-                var sqlStatementRaw = _fileService.ReadAllText(scriptFile);
-                var sqlStatements = _dataService.BreakStatements(sqlStatementRaw)
-                    .Where(s => !string.IsNullOrWhiteSpace(s))
+            sqlScriptFiles.Sort();
+            sqlScriptFiles.ForEach(scriptFile =>
+            {
+                var sqlStatementRaw = _fileService.ReadAllText(scriptFile);
+                var sqlStatements = _dataService.BreakStatements(sqlStatementRaw)
+                    .Where(s => !string.IsNullOrWhiteSpace(s))
                     .ToList();
 
-                sqlStatements.ForEach(sqlStatement =>
-                {
+                sqlStatements.ForEach(sqlStatement =>
+                {
                     try
                     {
                         sqlStatement = _tokenReplacementService.Replace(tokenKeyPairs, sqlStatement);
                         _traceService.Debug($"Executing sql statement as part of : {scriptFile}");
 
-                        _configurationDataService.ExecuteSql(
-                            connection: connection,
-                            commandText: sqlStatement,
-                            transaction: transaction,
-                            commandTimeout: commandTimeout,
-                            traceService: _traceService);
-                    }
-                    catch (Exception)
+                        _configurationDataService.ExecuteSql(connection: connection,
+                                                             commandText: sqlStatement,
+                                                             transaction: transaction,
+                                                             commandTimeout: commandTimeout,
+                                                             traceService: _traceService);
+                    } catch(Exception)
                     {
                         _traceService.Error($"Failed to execute sql statements in script file {scriptFile}.{Environment.NewLine}" +
                             $"The failing statement starts here --------------------------{Environment.NewLine}" +
                             $"{sqlStatement} {Environment.NewLine}" +
                             $"The failing statement ends here --------------------------");
                         throw;
-                    }
+                    }
                 });
 
-                _traceService.Info($"Executed script file {scriptFile}.");
-            });
-        }
+                _traceService.Info($"Executed script file {scriptFile}.");
+            });
+        }
+
+        public virtual void RunNonVersionDropScripts(IDbConnection connection,

```

Comment:
Why do you create a new RunNonVersionDropScripts? Is RunNonVersionScripts sufficient for this?